### PR TITLE
enable/fix make install

### PR DESCRIPTION
**Pull request by keerockl**
1. disable doxygen
2. add daemon_cl into bin
3. copy smartx.conf into alsa conf path( /etc/alsa/conf.d)

cmake -DIAS_IS_HOST_BUILD=1 -DCMAKE_INSTALL_PREFIX=/usr ../

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,21 +101,33 @@ if (${IAS_IS_HOST_BUILD})
   include(CMakeLists.avb_tests.txt)
 endif()
 
+#------------------------------------------------------------------
+#Install
+#------------------------------------------------------------------
+
+install(CODE "execute_process(COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/deps/gptp/linux/build/obj/daemon_cl ${CMAKE_INSTALL_PREFIX}/bin/daemon_cl)")
+install(CODE "execute_process(COMMAND mkdir -p /etc/alsa/conf.d)")
```
Comment (edersondisouza) [inline CMakeLists.txt]: Why not use install(FILES) or install(PROGRAMS)? Then you don't have to run the commands, cmake actually tracks the installed files, etc. See https://cmake.org/cmake/help/v3.0/command/install.html

File: CMakeLists.txt [hunk 2]
```diff
@@ -103,21 +101,33 @@ if (${IAS_IS_HOST_BUILD})
...
+
+install(CODE "execute_process(COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp)")
+install(CODE "execute_process(COMMAND sed -i s@<lib_path>@${CMAKE_INSTALL_LIBDIR}@g ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp)")
```
Comment (edersondisouza) [inline CMakeLists.txt]: Maybe use cmake configure_file (https://cmake.org/cmake/help/v3.0/command/configure_file.html) in place of this sed?

**Review by edersondisouza — CHANGES_REQUESTED** [review]
As the documentation change appears unrelated, I'd separate it on different commit (in this same PR).

File: CMakeLists.txt [hunk 3]
```diff
@@ -105,12 +105,19 @@ endif()
 #Install
 #------------------------------------------------------------------
 
-install(CODE "execute_process(COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/deps/gptp/linux/build/obj/daemon_cl ${CMAKE_INSTALL_PREFIX}/bin/daemon_cl)")
-install(CODE "execute_process(COMMAND mkdir -p /etc/alsa/conf.d)")
+install(PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/deps/gptp/linux/build/obj/daemon_cl DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
```
Comment (edersondisouza) [inline CMakeLists.txt]: Why not ${CMAKE_INSTALL_BINDIR}?

File: CMakeLists.txt [hunk 4]
```diff
@@ -105,12 +105,19 @@ endif()
...
-install(CODE "execute_process(COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp)")
-install(CODE "execute_process(COMMAND sed -i s@<lib_path>@${CMAKE_INSTALL_LIBDIR}@g ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp)")
-install(CODE "execute_process(COMMAND mv ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp /etc/alsa/conf.d/51-smartx.conf)")
+set(INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR}/alsa-lib)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.in ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf @ONLY)
+
+if(ASOUND_VERSION VERSION_LESS 1.1.7)
+        set(ALSA_CONF_PATH "/usr/share/alsa/alsa.conf.d")
+else()
+        install(DIRECTORY DESTINATION /etc/alsa/conf.d)
```
Comment (edersondisouza) [inline CMakeLists.txt:116]: Is it really necessary to create this dir?

Comment (keerockl) [reply CMakeLists.txt:116]: yes

File: CMakeLists.txt (hunk 3 above)
Comment (keerockl) [reply CMakeLists.txt]: should be the same but will switch to ${CMAKE_INSTALL_BINDIR}

File: public/res/51-smartx.conf [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  @args.NAME {
+    type string
+  }
+  type smartx
+  name $NAME
+}
+
+pcm_type.smartx
+{
+	lib "<lib_path>/libasound_module_pcm_smartx.so"
```
Comment (edersondisouza) [inline public/res/51-smartx.conf]: Why not add here with the `@@` tag? So it's one less change on the next patch. And also add with the right name...

**Review by edersondisouza — COMMENTED** [review]
Maybe squash everything, and then separate the "disabling doxygen" part on a new patch?

File: public/res/51-smartx.conf (hunk 5 above)
Comment (keerockl) [reply public/res/51-smartx.conf]: oops. I am not supposed to push this file. let me check

Comment (keerockl) [reply public/res/51-smartx.conf]: oh you were looking at the outdated file. it's changed into 51-smartx.conf.in file.

**Review by edersondisouza — APPROVED** [review]

**Pull request merged**